mlflow/langchain/runnables.py

```diff
@@ -395,34 +399,68 @@ def _save_runnable_assign(model, file_path, loader_fn=None, persist_dir=None):
     _save_runnable_with_steps(model.mapper, mapper_path, loader_fn, persist_dir)
 
 
-def _save_picklable_runnable(model, path):
+def is_basic_types(value):
+    return value is None or isinstance(value, (str, int, float, bool))
+
+
+def _save_runnable_binding(model, file_path, loader_fn=None, persist_dir=None):
+    save_path = Path(file_path)
+    save_path.mkdir(parents=True, exist_ok=True)
+    model_config = {}
+
+    # runnableBinding bound is the real runnable to be invoked
+    model_config["bound"] = _save_runnables(model.bound, save_path, loader_fn, persist_dir)
+
+    # save other fields
+    for field, value in model.dict().items():
+        if is_basic_types(value):
```

Comment:
any chance that values is a list or an object?

Reply:
Yes, but is it necessary?

Reply:
can we use `json.dumps` instead to check json-serializability?

Reply:
Had a test, we can't.. Even on bound runnable, json.dumps works 🤣 But we can't load it back